backend/internal/service/matching/execute.go

```diff
@@ -0,0 +1,110 @@
+package matching
+
+import "day-trading-app/backend/internal/service/models"
+
+var s serviceImpl
+
+func ExecuteOrders(stockTxCommitQueue []models.StockMatch) {
+	for _, tx := range stockTxCommitQueue {
+		if isParent(tx) {
+			// Update stock transaction as completed
+			s.db.UpdateStockOrderStatus(tx.Order.UserName, tx.Order.StockTxID, tx.Order.OrderStatus)
+		} else { // child or non-parent
+			if tx.Order.IsBuy {
+				executeBuy(tx)
+			} else {
+				executeSell(tx)
+			}
+		}
+	}
+}
+
+func executeBuy(tx models.StockMatch) {
+	if tx.Order.OrderStatus == "IN_PROGRESS" { // unfulfilled
+		// Add deducted money back to wallet
+		var deducted = tx.Order.Quantity * tx.Order.StockPrice
+		walletBalance, _ := s.db.GetWalletBalance(tx.Order.UserName)
+		s.db.SetWalletBalance(tx.Order.UserName, walletBalance+deducted)
+
+		// Delete stock transaction todo
+
+		// Delete wallet transaction todo
+
+	} else if tx.Order.OrderStatus == "PARTIALLY_FULFILLED" {
+		// Refund remaining wallet amount
+		var deducted = tx.Order.Quantity * tx.Order.StockPrice
+		walletBalance, _ := s.db.GetWalletBalance(tx.Order.UserName)
+		s.db.SetWalletBalance(tx.Order.UserName, walletBalance+(deducted-tx.CostTotalTx))
+
+		// Delete wallet transaction todo
+
+	} else if tx.Order.OrderStatus == "COMPLETED" {
+		// Add stock quantity to user portfolio
+		s.db.AddStockToUser(tx.Order.UserName, tx.Order.StockID, tx.Order.Quantity)
+
+		// Update stock transaction status to completed
+		s.db.UpdateStockOrderStatus(tx.Order.UserName, tx.Order.StockTxID, tx.Order.OrderStatus)
+
+	} else if tx.Order.OrderStatus == "" {
+		// Error: empty string
+	} else {
+		// Error: spelling probably
+	}
+}
+
+func executeSell(tx models.StockMatch) {
+	if tx.Order.OrderStatus == "IN_PROGRESS" { // unfulfilled
+		// Add stock quantity back to user portfolio
+		s.db.AddStockToUser(tx.Order.UserName, tx.Order.StockID, tx.Order.Quantity)
+
+		// Delete stock transaction // todo
+
+	} else if tx.Order.OrderStatus == "PARTIALLY_FULFILLED" {
+		// Add remaining stock quantity back to user portfolio (remaining = Order.Quantity - QuantityTx)
+		s.db.AddStockToUser(tx.Order.UserName, tx.Order.StockID, tx.Order.Quantity-tx.QuantityTx)
+
+	} else if tx.Order.OrderStatus == "COMPLETED" {
+		// Update stock transaction to completed
+		s.db.UpdateStockOrderStatus(tx.Order.UserName, tx.Order.StockTxID, tx.Order.OrderStatus)
+
+		// Add money to wallet
+		walletBalance, _ := s.db.GetWalletBalance(tx.Order.UserName)
+		s.db.SetWalletBalance(tx.Order.UserName, walletBalance+tx.CostTotalTx)
+
+		// Insert wallet transaction todo
+
+	} else if tx.Order.OrderStatus == "" {
+		// Error: empty string
+	} else {
+		// Error: spelling probably
+	}
+}
+
+func isParent(tx models.StockMatch) bool {
+	return (tx.Order.OrderStatus != "IN_PROGRESS") && (tx.PriceTx == 0)
+}
+
+type Database interface {
```

Comment:
this is not how you import the DB handler into another package. This wont work and needs to be deleted

Reply:
Actually give me a list of things you want to manipulate in the database, like add something, update something, delete something. I can create that function for you. I feel like we should have only the store layer talking to the db cuz later if I want to make changes to our data model, we’d have to adjust engine/execution and store layer which is not efficient.